components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/model/FlowContext.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.execution.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FlowContext {
+
+    private final Map<String, Object> contextData = new HashMap<>();
+
+    private FlowContext() {
+
+    }
+
+    public Map<String, Object> getContextData() {
+
+        return contextData;
```

Comment:
```suggestion
        return Collections.unmodifiableMap(contextData);
```

Reply:
As this data is set from the consumer and used by the respective downstream component IMO, it's not important to make this unmodifiable